metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -172,6 +180,66 @@ public TableDto create(final QualifiedName name, final TableDto tableDto) {
         return dto;
     }
 
+    /**
+     * Logs diagnostic data for debugging invalid owners.
+     *
+     * @param name the qualified name of the resource
+     * @param tableDto the table details after all processing has been done on the owner
+     */
+    private void logOwnershipDiagnosticDetails(final QualifiedName name, final TableDto tableDto) {
+        try {
+            final String tableOwner = Optional.ofNullable(tableDto.getDefinitionMetadata())
+                                          .map(node -> node.get("owner"))
+                                          .map(owner -> owner.get("userId"))
+                                          .map(JsonNode::textValue)
+                                          .orElse(null);
+
+            if (tableOwner == null || "metacat".equals(tableOwner) || "root".equals(tableOwner)) {
```

Comment:
Should this also check for empty string

Reply:
Although unlikely probably safer to add. Added.